vita3k/modules/SceKernelThreadMgr/SceThreadmgr.cpp

```diff
@@ -336,8 +336,59 @@ EXPORT(int, _sceKernelGetMutexInfo, SceUID mutexId, SceKernelMutexInfo *pInfo) {
     return SCE_KERNEL_OK;
 }
 
-EXPORT(int, _sceKernelGetRWLockInfo) {
-    TRACY_FUNC(_sceKernelGetRWLockInfo);
+EXPORT(int, _sceKernelGetRWLockInfo, SceUID rwlockId, SceKernelRWLockInfo *info) {
+    TRACY_FUNC(_sceKernelGetRWLockInfo, rwlockId, info);
+    if (!info)
+        return RET_ERROR(SCE_KERNEL_ERROR_INVALID_ARGUMENT);
+    if (info->size < sizeof(SceKernelRWLockInfo))
+        return RET_ERROR(SCE_KERNEL_ERROR_INVALID_ARGUMENT);
+    const RWLockPtr rwlock = lock_and_find(rwlockId, emuenv.kernel.rwlocks, emuenv.kernel.mutex);
+    if (!rwlock)
+        return RET_ERROR(SCE_KERNEL_ERROR_UNKNOWN_RW_LOCK_ID);
+    const std::lock_guard<std::mutex> rwlock_lock(rwlock->mutex);
+    info->rwLockId = rwlock->uid;
+    std::copy(rwlock->name, rwlock->name + KERNELOBJECT_MAX_NAME_LENGTH, info->name);
```

Comment:
```suggestion
   strncpy(info->name, rwlock->name, KERNELOBJECT_MAX_NAME_LENGTH)
```
Look cleaner and if the name has size KERNELOBJECT_MAX_NAME_LENGTH, the trailing zero will be missing.

Reply:
I did the same for all other same cases